### PR TITLE
feat: added nameable

### PR DESCRIPTION
**Pull request by HeDevedUp**
## Description

**Task:** https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_people.md

This PR makes the following changes:
- Creates a base Decorator class, which inherits from Nameable.
-  Creates the Nameable class, which the Person class inherits from.
-  Creates the CapitalizeDecorator and TrimmerDecorator classes which inherit from Decorator.

## Type of change 
- [X] refactored the code base and added a component folder
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented on my code
- [X] I have deleted any unused code as comments in my codebase

### CODE REVIEW

File: decorator/trim_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative 'decorator'
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
+  end
```
Comment (uwadonat) [inline decorator/trim_decorator.rb]: - [ ] Let's trim the word only if the name has a maximum of 10 character.
`Hint`: you should use [ternary operator](https://www.rubyguides.com/2019/10/ruby-ternary-operator/) to achieve this.

File: README.md [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 ## Authors
 
-👤 ** J Uzoma **
+👤 ** JESSE  Uzoma **
 
 - GitHub: [@HedevedUp](https://github.com/HeDevedUp)
 - Twitter: [devtochi](https://twitter.com/devtochi)
```
Comment (uwadonat) [inline README.md:30]: - [ ] Let's add instructions to run our project. This will make it more user friendly and professional.

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @HeDevedUp,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your commit messages are very descriptive ✔️
- Descriptive pull request message ✔️
- Well organized and clean project ✔️
almoo

## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @HeDevedUp 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job Implimenting the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**